### PR TITLE
Enhance feat query string

### PR DESCRIPTION
**Pull request by rickyschools**
enable feature to allow users to pass `num_filings_to_download` to count parameter in `form_query_string`

### CODE REVIEW

**Comment by jadchaar** [comment]
Hey @rickyschools, thanks for the contribution! Are you looking to pass the count value into the query string because you are looking for >100 filings?

Sadly, SEC Edgar system currently does not support the passing of >100 into the count QS parameter. The system officially supports 10, 20, 40, 80, and 100 entries, and any values in between will round down. For example, if you put a count of 50, it will return 40 values.

For a full example example, this search URL `https://www.sec.gov/cgi-bin/browse-edgar?action=getcompany&CIK=0000320193&type=&dateb=20190527&owner=exclude&start=0&count=500&output=atom`, which has the `count=500`, only returns an XML document with 100 entries, regardless of the value passed into count.

That said, the SEC Edgar system does have support in place for non-standard count values if you pass in the `start` parameter to paginate the results. For example, if you have 53 results, you'd need to pass in a URL with `count=40` then another `count=40` to get the remaining 13. It is a pain, but you need to modulo the value by the closest "supported" value (e.g. 150 results would need 2 counts of 100 or 80, but 100 may be easier for implementation reasons).

If you'd like to implement support for the start parameter to get more counts supported, I'd love to review it :). It is on the project roadmap though, so I am aware of the issue :).

**Comment by rickyschools** [comment]
@jadchaar - I'm sorry for the delayed response and the random PR. Candidly, I was perusing a few repos that extracted information from SEC/Edgar to see how specific form types were handled/parsed. In reviewing your code, I saw the # todo comments and thought I'd help out and it'd be a more straight-forward commit; albeit I should have done more due diligence and further researched/tested before I sent along. 

Thanks for your detailed response above. In a past life I was a bit more familiar with the Edgar system and your comments jogged the memories surrounding the export limitations. 

I'm not sure if you're familiar with, or have explored, directory browsing via Edgar. It largely removes the issues surrounding pagination, but introduce other _issues_ like being naive to filing types and needing to _peak in_ each of the downloaded forms (to extract the relative filing type, if necessary). https://www.sec.gov/edgar/searchedgar/accessing-edgar-data.htm

TL;DR
You can pass a CIK to a URL and retrieve all the accession numbers filed, via json, by following the below url structure. 
`https://www.sec.gov/Archives/edgar/data/{cik-id}/index.json`

I'm happy to help you further your libraries capabilities, but have built what I needed via the method I described above. 

All the best, 

RS

**Comment by jadchaar** [comment]
Thanks for the follow-up @rickyschools, I really appreciate the extra info.

So the method you are following is to retrieve the `index.json` file with all the accession numbers and to bulk download all accessions? Or are you fetching each accession and seeing if the filing type is what you are looking for? I am curious to understand the approach you are taking to see if it could be use to this library :).

**Comment by rickyschools** [comment]
I've enabled optionality for both the scenarios you describe. The short is I have a directory 'scraper' (which is really a directory crawler) that grabs/builds access points (accession urls) for a given CIK. Each are turned into a filing object that have download/scrape functionality where a user can pass the filing types they desire to download. As such, we're now operating serially on single filings at a time - but now we have no download limits ('bulk' downloading). 

Let me know if you're interested in discussing further and I'll send along my email. 

RS
